### PR TITLE
test(e2e): migrate `oras tag` specs to ZOT

### PR DESCRIPTION
**Pull request by qweeah**
Depends on merging of https://github.com/oras-project/oras/pull/1086

### CODE REVIEW

**Review by Wwwsylvia — APPROVED** [review]
LGTM, but I am not a maintainer

File: test/e2e/suite/command/tag.go [hunk 1]
```diff
@@ -75,51 +75,29 @@ var _ = Describe("OCI image layout users:", func() {
...
-		hint := regexp.QuoteMeta(fmt.Sprintf("Tagging [oci-layout] %s", LayoutRef(root, digest)))
-		gomega.Expect(out).To(gbytes.Say(hint))
-		gomega.Expect(out).NotTo(gbytes.Say(hint)) // should only say hint once
-		ORAS("repo", "tags", Flags.Layout, LayoutRef(root, "")).MatchKeyWords(tags...).Exec()
-	}
-	When("running `tag`", func() {
-		prepare := func() string {
-			root := GinkgoT().TempDir()
-			ORAS("cp", RegistryRef(Host, ImageRepo, multi_arch.Tag), Flags.ToLayout, LayoutRef(root, multi_arch.Tag)).Exec()
-			return root
```
Comment (Wwwsylvia) [inline test/e2e/suite/command/tag.go:110]: Do we not need this?

Comment (qweeah) [reply test/e2e/suite/command/tag.go:110]: because we are using ZOT and can copy from oci layout directly

Comment (Wwwsylvia) [reply test/e2e/suite/command/tag.go:110]: How about this part?

```go
ORAS("repo", "tags", Flags.Layout, LayoutRef(root, "")).MatchKeyWords(tags...).Exec()
```

Comment (qweeah) [reply test/e2e/suite/command/tag.go:110]: see https://github.com/oras-project/oras/blob/main/test/e2e/internal/utils/prepare.go#L41

Comment (Wwwsylvia) [reply test/e2e/suite/command/tag.go:110]: OK this part is not actually removed. The git diff rendering is kinda misleading.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**